### DISCUSSION TITLE
Unique item ids

### DISCUSSION THREAD

**Discussion by venkatramanm** [beckn/protocol-specifications#178] [Open Commerce Networks]
To make sure that item_ids returned by bpps are unique in the network. It is important to come up with some scheme for mapping internal ids to beckn-network-unique-item-ids using simple  transformations such as ./[domain_name]/[internal_id]@[subscriber_id]. This would needed so that participants can handle uniqueness easily.

**Comment by venkatramanm** [comment]
Id Uniqueness across network may be needed for multiple entities. Like fulfilliment,category,provider, provider_category,order, catalog, item, cancellation/return reason.....

./[domain_name]/[internal_item_id]@[provider_id]@[subscriber_id] or some such scheme should work to make it unique. 

@core-wg-admin comment from slack "Good point. We can definitely consider something to uniquely identify items. I quite like the URL scheme used by upi, payto etc. Something like item://id/12577?provider. id=abcd what do you think?"

**Comment by ravi-prakash-v** [comment]
Let's look at it from the other way. If such id scheme doesn't exist. Would it be an issue?

**Reply by ravi-prakash-v** [reply]
Just applying the principle of optimal ignorance. Basically, a) do you need it? and b) do you need it now?

**Reply by venkatramanm** [reply]
For usecases like rating, if we want to rate a product, provider, order, user .. with out the I'd being uniq, we will need to pass a lot more attributes in the rating schema to make it uniq.

Same with search. Even if you have item id, you will need to pass a lot of information to make it unique. It is.counter intuitive that an "Id" field is not uniq . I am ok to pass more attibutes but that would mean , schema update for rating and possibly in other places. 🤔
